File: 1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.rb
#* 1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold
arr = [2,2,2,2,5,5,5,8]; k = 3; threshold = 4
# @param {Integer[]} arr
# @param {Integer} k
# @param {Integer} threshold
# @return {Integer}
#* SOL 2 | CONCISE
def num_of_subarrays(arr, k, threshold)
  (0..arr.size-k).count do |i|
    arr[i...i+k].sum / 3 >= threshold
  end
end
p num_of_subarrays(arr, k, threshold)

#* SOL 1
def num_of_subarrays(arr, k, threshold)
  count=0
  (0..arr.size-k).each do |i|
    p arr[i...i+k]
    p arr[i...i+k].sum / 3 >= threshold
    count+=1 if arr[i...i+k].sum / 3 >= threshold
  end
  count
end
p num_of_subarrays(arr, k, threshold)

File: 1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.js
// JAVASCRIPT
// 1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold
arr = [2,2,2,2,5,5,5,8]; k = 3; threshold = 4

//SOL
function numOfSubarrays(arr, k, threshold) {
    let count = 0;
    for (let i = 0; i <= arr.length - k; i++) {
        let sum = 0;
        for (let j = i; j < i + k; j++) {
            sum += arr[j];
        }
        if (sum / k >= threshold) {
            count++;
        }
    }
    return count;
}
console.log(numOfSubarrays([2, 1, 3, 4, 1], 3, 2));


